src/engraving/layout/v0/layoutcontext.cpp

```diff
@@ -32,209 +32,382 @@
 {
     IF_ASSERT_FAILED(m_score) {
-        return 0;
+        static const SpannerMap dummy;
+        return dummy;
     }
-    return m_score->nstaves();
+    return m_score->spannerMap();
 }
 
-const std::vector<Staff*>& LayoutContext::staves() const
+const ChordRest* DomAccessor::findCR(Fraction tick, track_idx_t track) const
 {
     IF_ASSERT_FAILED(m_score) {
-        static const std::vector<Staff*> dummy;
-        return dummy;
+        return nullptr;
     }
-    return m_score->staves();
+    return m_score->findCR(tick, track);
 }
 
-const Staff* LayoutContext::staff(staff_idx_t idx) const
+ChordRest* DomAccessor::findCR(Fraction tick, track_idx_t track)
 {
     IF_ASSERT_FAILED(m_score) {
         return nullptr;
     }
-    return m_score->staff(idx);
+    return m_score->findCR(tick, track);
 }
 
-size_t LayoutContext::ntracks() const
+MeasureBase* DomAccessor::first()
 {
     IF_ASSERT_FAILED(m_score) {
-        return 0;
+        return nullptr;
     }
-    return m_score->ntracks();
+    return m_score->first();
 }
 
-const Measure* LayoutContext::tick2measure(const Fraction& _tick) const
+RootItem* DomAccessor::rootItem() const
 {
     IF_ASSERT_FAILED(m_score) {
         return nullptr;
     }
-    return m_score->tick2measure(_tick);
+    return m_score->rootItem();
 }
 
-compat::DummyElement* LayoutContext::dummyParent() const
+compat::DummyElement* DomAccessor::dummyParent() const
 {
     IF_ASSERT_FAILED(m_score) {
         return nullptr;
     }
     return m_score->dummy();
 }
 
-void LayoutContext::undoAddElement(EngravingItem* item, bool addToLinkedStaves, bool ctrlModifier)
+void DomAccessor::undoAddElement(EngravingItem* item, bool addToLinkedStaves, bool ctrlModifier)
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->undoAddElement(item, addToLinkedStaves, ctrlModifier);
 }
 
-void LayoutContext::undoRemoveElement(EngravingItem* item)
+void DomAccessor::undoRemoveElement(EngravingItem* item)
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->undoRemoveElement(item);
 }
 
-void LayoutContext::undo(UndoCommand* cmd, EditData* ed) const
+void DomAccessor::undo(UndoCommand* cmd, EditData* ed) const
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->undo(cmd, ed);
 }
 
-void LayoutContext::addElement(EngravingItem* item)
+void DomAccessor::addElement(EngravingItem* item)
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->addElement(item);
 }
 
-void LayoutContext::removeElement(EngravingItem* item)
+void DomAccessor::removeElement(EngravingItem* item)
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->removeElement(item);
 }
 
-void LayoutContext::addUnmanagedSpanner(Spanner* s)
+void DomAccessor::addUnmanagedSpanner(Spanner* s)
 {
     IF_ASSERT_FAILED(m_score) {
         return;
     }
     m_score->addUnmanagedSpanner(s);
 }
 
+const std::set<Spanner*> DomAccessor::unmanagedSpanners()
```

Comment:
It should probably be const std::set<Spanner*>&

Reply:
Yes, it is, but at the moment I did the same as score 
https://github.com/musescore/MuseScore/blob/master/src/engraving/libmscore/score.h#L1157
very strange, not a const method that returns a copy... 
I have a bookmark at this place, I plan to fix it separately (as part of the task of cleaning the score interface)